File: src/main/java/com/bdev/hogwarts_api/service/teacher/TeacherService.java
package com.bdev.hogwarts_api.service.teacher;

import com.bdev.hogwarts_api.data.dto.teacher.Teacher;

import java.util.List;
import java.util.Optional;

public interface TeacherService {
    boolean exists(long id);
    List<Teacher> getAllTeachers();
    Optional<Teacher> getTeacherById(long id);
    Optional<Teacher> getTeacherByLogin(String login);
    long createTeacher(Teacher teacher);
    void editTeacher(Teacher teacher);
    void deleteTeacher(long id);
}

File: src/main/java/com/bdev/hogwarts_api/service/teacher/TeacherService.kt
package com.bdev.hogwarts_api.service.teacher

import com.bdev.hogwarts_api.data.dto.teacher.Teacher

interface TeacherService {
    fun getAllTeachers(): List<Teacher>
    fun exists(id: Long): Boolean
    fun getTeacherById(id: Long): Teacher?
    fun getTeacherByLogin(login: String): Teacher?
    fun createTeacher(teacher: Teacher): Long
    fun editTeacher(teacher: Teacher)
    fun deleteTeacher(id: Long)
}
